WalletWasabi/WabiSabi/Client/Banning/CoinPrison.cs

```diff
@@ -0,0 +1,99 @@
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq;
+using WalletWasabi.Blockchain.TransactionOutputs;
+using WalletWasabi.Helpers;
+using WalletWasabi.Logging;
+
+namespace WalletWasabi.WabiSabi.Client.Banning;
+
+public class CoinPrison
+{
+	public CoinPrison(string filePath)
+	{
+		FilePath = filePath;
+	}
+
+	public List<PrisonedCoinRecord> BannedCoins { get; set; } = new();
+	public string FilePath { get; set; }
+
+	public bool TryGetOrRemoveBannedCoin(SmartCoin coin, DateTimeOffset when, [NotNullWhen(true)] out DateTimeOffset? bannedUntil)
```

Comment:
Yes, good argument. @lontivero suggested that it could get the end of input registration so we know if the coin can be registered as an input for the next coinjoin, but I don't see where to get that timestamp from. Also feels like an irrelevant nit, but it might just be me.